packages/vite/src/node/server/transformRequest.ts

```diff
@@ -457,8 +457,9 @@ async function handleModuleSoftInvalidation(
   }
 
   let result: TransformResult
-  // For SSR soft-invalidation, no transformation is needed
-  if (transformResult.ssr) {
+  // For SSR or direct CSS, no transformation is needed
+  // Note that direct CSS soft invalidation happens when files included by tailwind's `content` option is updated
+  if (transformResult.ssr || mod.type === 'css') {
```

Comment:
Should the fix be applied to https://github.com/vitejs/vite/blob/1fe949d56f5179aa1e74e2edc94c600f93a1180b/packages/vite/src/node/server/moduleGraph.ts#L218-L219 instead so that we don't soft-invalidate if the importer if is `type === 'css'`?

That way the CSS is re-transformed. Otherwise if we fix it here, we're reusing the old `transformResult`. I tried it and it also fixes it for me locally. However, I can't quite test that the re-transform issue is happening though, it seems like either way tailwindcss can update it normally. So maybe not a big deal to apply my fix?

Reply:
That sounds correct. My fix did not have a problem for tailwind because tailwind only needs to be updated when the file included in the `content` option and
those files have the `tailwind.css` file in the `importer` but not in `staticImportedUrls`, resulting `tailwind.css` file to be hard invalidated when needed.
https://github.com/vitejs/vite/blob/cb40cb55525e2051ac4f6e341511ce4bb23e4bfe/packages/vite/src/node/server/moduleGraph.ts#L218-L220

Reply:
Updated 👍